### PR TITLE
Cleanup API in README

### PR DESCRIPTION
**Pull request by steelbrain**
Just like the title says

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
-  static find(directory:String, names: String | Array<string>): ?String
-  static findCached(directory:String, names: String | Array<string>): ?String
-  static findAsync(directory: String, names: String | Array<string>): Promise<?String>
-  static findCachedAsync(directory: String, names: String | Array<string>): Promise<?String>
-  static tempFile<T>(fileName:String, fileContents:String, Callback:Function<T>):Promise<T>
-  static tempFiles<T>(filesNames:Array<{name: String, contents: String}>, callback:Function<T>):Promise<T>
-}
+export function exec(command: String, args: Array<string> = [], options: Object)
+export function execNode(filePath: String, args: Array<string> = [], options: Object)
+export function parse(data: String, regex: String, options: Object = {baseReduction: 1, flags: ""})
```
Comment (Arcanemagus) [inline README.md]: `baseReduction` isn't used in the code anywhere, where did this come from?

Comment (Arcanemagus) [inline README.md]: Also shouldn't this show that by default (forced) the `g` flag is used?

Comment (steelbrain) [inline README.md]: It was used to convert `Line 5` to `Line 4` make it Atom's 0-based range. Done

File: README.md [hunk 2]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
+export function rangeFromLineNumber(textEditor: TextEditor, lineNumber: Number, colStart: Number = <firstColumn>):Array
```
Comment (Arcanemagus) [inline README.md]: `lineNumber` is optional, defaulting to `0`.

`fistColumn` -> `firstTextColumn`? Not sure this is necessary or not.

Comment (steelbrain) [inline README.md]: Done

File: README.md [hunk 3]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
-  static parse(data: String, regex: String, options: Object = {baseReduction: 1, flags: ""})
-  static rangeFromLineNumber(textEditor: TextEditor, lineNumber: Number, colStart: Number = <firstColumn>):Array
-  static find(directory:String, names: String | Array<string>): ?String
-  static findCached(directory:String, names: String | Array<string>): ?String
-  static findAsync(directory: String, names: String | Array<string>): Promise<?String>
-  static findCachedAsync(directory: String, names: String | Array<string>): Promise<?String>
-  static tempFile<T>(fileName:String, fileContents:String, Callback:Function<T>):Promise<T>
-  static tempFiles<T>(filesNames:Array<{name: String, contents: String}>, callback:Function<T>):Promise<T>
-}
+export function exec(command: String, args: Array<string> = [], options: Object)
```
Comment (Arcanemagus) [inline README.md]: Should the return type be specified on these functions that return something?

Comment (steelbrain) [inline README.md]: Added return types for `exec`, `execNode` and `parse`

**Comment by steelbrain** [comment]
**Note:** CI build is failing because of https://github.com/babel/babel-eslint/issues/267

File: README.md [hunk 4]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
-  static findAsync(directory: String, names: String | Array<string>): Promise<?String>
-  static findCachedAsync(directory: String, names: String | Array<string>): Promise<?String>
-  static tempFile<T>(fileName:String, fileContents:String, Callback:Function<T>):Promise<T>
-  static tempFiles<T>(filesNames:Array<{name: String, contents: String}>, callback:Function<T>):Promise<T>
-}
+export function exec(command: String, args: Array<string> = [], options: Object): Promise
+export function execNode(filePath: String, args: Array<string> = [], options: Object): Promise
+export function parse(data: String, regex: String, options: Object = {flags: 'g'}): Array<Linter$Message>
+export function rangeFromLineNumber(textEditor: TextEditor, lineNumber: Number = 0, colStart: Number = <firstTextColumn>):Array
+export function find(directory:String, names: String | Array<string>): ?String
```
Comment (Arcanemagus) [inline README.md]: `directory:String` -> `directory: String`

File: README.md [hunk 5]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
+export function findCached(directory:String, names: String | Array<string>): ?String
```
Comment (Arcanemagus) [inline README.md]: `directory:String` -> `directory: String`

File: README.md [hunk 6]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
+export function findAsync(directory: String, names: String | Array<string>): Promise<?String>
+export function findCachedAsync(directory: String, names: String | Array<string>): Promise<?String>
+export function tempFile<T>(fileName:String, fileContents:String, Callback:Function<T>):Promise<T>
```
Comment (Arcanemagus) [inline README.md]: Spaces before the types here as well.

File: README.md [hunk 7]
```diff
@@ -10,18 +10,16 @@ For full documentation of `exec` and `execNode` API, please refer to [`sb-exec R
...
+export function tempFiles<T>(filesNames:Array<{name: String, contents: String}>, callback:Function<T>):Promise<T>
```
Comment (Arcanemagus) [inline README.md]: And here.

File: README.md (hunk 4 above)
Comment (steelbrain) [inline README.md]: There should be a linter that does that, I had previously just copy/pasted, I'll tweak them all now

**Comment by Arcanemagus** [comment]
I think that last batch of comments is the last thing I see.

**Comment by steelbrain** [comment]
All comments taken care of

**Pull request merged**